iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/selector/impl/SizeTieredCompactionSelector.java

```diff
@@ -165,12 +170,17 @@ public List<List<TsFileResource>> selectInnerSpaceTask(List<TsFileResource> tsFi
     PriorityQueue<Pair<List<TsFileResource>, Long>> taskPriorityQueue =
         new PriorityQueue<>(new SizeTieredCompactionTaskComparator());
     try {
-      int maxLevel = searchMaxFileLevel();
-      for (int currentLevel = 0; currentLevel <= maxLevel; currentLevel++) {
-        if (!selectLevelTask(currentLevel, taskPriorityQueue)) {
-          break;
+      selectMaxModsFileTask(taskPriorityQueue);
```

Comment:
One suggestion is that it is better to make every method as `pure function`, that is, do not pass the variable `taskPriorityQueue` into the method and add element into it inside the method. 

We can change it to something like `taskPriorityQueue.addAll(selectMaxModsFileTask())`